lib/optimizely/config_manager/http_project_config_manager.rb

```diff
@@ -0,0 +1,214 @@
+# frozen_string_literal: true
+
+#
+#    Copyright 2019, Optimizely and contributors
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+require_relative '../config/datafile_project_config'
+require_relative '../error_handler'
+require_relative '../exceptions'
+require_relative '../helpers/constants'
+require_relative '../logger'
+require_relative '../notification_center'
+require_relative '../project_config'
+require_relative 'project_config_manager'
+require_relative 'async_scheduler'
+require 'httparty'
+require 'json'
+module Optimizely
+  class HTTPProjectConfigManager < ProjectConfigManager
+    # Config manager that polls for the datafile and updated ProjectConfig based on an update interval.
+
+    attr_reader :config
+
+    # Initialize config manager. One of sdk_key or url has to be set to be able to use.
+    #
+    # sdk_key - Optional string uniquely identifying the datafile.
```

Comment:
Let's also note here that the `sdk_key` is required unless a `url` is passed in.